### ISSUE TITLE
Request to Implement XDG Base Directory Specification

### ISSUE THREAD

**Issue by rwnobrega** [otfried/ipe#513]
Currently, in Linux systems, Ipe stores its configuration files in the `~/.ipe/` directory. This location does not conform to the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html).

To improve compatibility and organization, the Ipe configuration files should be moved to `$XDG_CONFIG_HOME/ipe/`, where the `$XDG_CONFIG_HOME` environment variable defaults to `~/.config/`.

Thank you for your work on Ipe and for considering this request.

**Comment by otfried** [comment]
I'm not really sure if what is in `~/.ipe` qualifies as configuration files.

The contents of `~/.ipe/latexrun` should most likely go in `$XDG_CACHE_HOME/ipe` or perhaps `$XDG_CACHE_HOME/ipe/latexrun`.

In addition, there are ipelets (which may include customization files) and style sheets.   My feeling is that ipelets and style sheets are data files, not configuration files. So perhaps the right organization is
```
$XDG_DATA_HOME/ipe/ipelets
$XDG_DATA_HOME/ipe/styles
$XDG_DATA_HOME/ipe/scripts
$XDG_CONFIG_HOME/ipe/customization.lua
$XDG_CONFIG_HOME/ipe/ipe.conf
$XDG_CACHE_HOME/ipe/latexrun
$XDG_CACHE_HOME/ipe/icons
$XDG_STATE_HOME/ipe/recent_files.lua
```
That's a lot of directories.

**Comment by rwnobrega** [comment]
Thank you for your response!

I agree with your proposed organization. Indeed, that's a lot of directories, but (in my opinion) it will be beneficial to Linux users.

**Comment by otfried** [comment]
Notes:
Extend `Platform::ipeDir` to handle all folders used by Ipe, on all platforms, for all configurations (with and without IPEBUNDLE).
First argument can be one of
```
lua
styles
user-styles
ipelets
user-ipelets
scripts
user-scripts
config
latex
icons
doc
state
```

Expose this function to Lua, and use it instead of `os.getenv` from Lua and instead of the `config` table.

**Comment by N-Coder** [comment]
Maybe instead of the `user-` versions just return a list of directories that shall be searched in order and where the first entry represents the default location to put new files. That way we could keep compatibility with the old layout.

If you want to save a little on distinct directories, Stylesheets and ipelets are probably sufficiently configuration-like to also store/search for them in the config directory. That might also make them slightly easier to find, as the data dir is in imho something rather arcane...

**Comment by otfried** [comment]
Implemented in 7.3.1.
